content/blog/cloud-native-secret-management-with-pulumi-esc-and-external-secrets-operator/index.md

```diff
@@ -0,0 +1,450 @@
+```
+
+### Step 3: Deploy an Application and Reference the Secret
+
+Now, we can deploy an application that references the secret from the Kubernetes cluster. I am going to deploy the app
+called `PodInfo` which is a simple application that exposes the pod information as an API. Add the following code to the
+`index.ts`:
+
+```typescript
+import * as k8s from "@pulumi/kubernetes";
+import * as pulumi from "@pulumi/pulumi";
+
+// cut for brevity
+const podInfo = new k8s.helm.v3.Release("podinfo", {
+    chart: "podinfo",
+    version: "6.7.0",
+    namespace: "podinfo",
+    createNamespace: true,
+    repositoryOpts: {
+        repo: "https://stefanprodan.github.io/podinfo",
+    },
+    values: {
+        extraEnvs: [
+            {
+                name: "FROM_ESC_VIA_ESO",
+                valueFrom: {
+                    secretKeyRef: {
+                        name: "esc-secret-store",
+                        key: "hello-secret",
+                    }
+                }
+            },
+            {
+                name: "FROM_ESC_VIA_ESO_2",
+                valueFrom: {
+                    secretKeyRef: {
+                        name: "esc-secret-store",
+                        key: "hello",
+                    }
+                }
+            }
+        ]
+    },
+});
+
+const externalSecretPodInfo = new k8s.apiextensions.CustomResource("external-secret-podinfo", {
+    apiVersion: "external-secrets.io/v1beta1",
+    kind: "ExternalSecret",
+    metadata: {
+        name: "esc-secret-store",
+        namespace: podInfo.namespace,
+    },
+    spec: {
+        dataFrom: [
+            {
+                extract: {
+                    conversionStrategy: "Default",
+                    key: "app",
+                }
+            }
+        ],
+        refreshInterval: "10s",
+        secretStoreRef: {
+            kind: clusterSecretStore.kind,
+            name: clusterSecretStore.metadata.name,
+        }
+    },
+});
+```
+
+This will deploy the `PodInfo` application and create an `ExternalSecret` resource that references the secret from the
+Pulumi ESC project. We mount the secret into the pod as an environment variable by using the `extraEnvs` field in the
+Helm chart.
+
+{{% notes type="info" %}}
+
+In your real-world application, you could add different ways to reference the secret, like mounting it as a volume or
+using `fromEnv` field in the Helm chart. It depends on your application and how you want to consume the secret.
+
+{{% /notes %}}
+
+After deploying the stack, you should see that the `PodInfo` application is running and if you port-forward the service
+by running:
+
+```bash
+kubectl port-forward svc/podinfo 9898:9898 -n podinfo
+```
+
+You can access the API by running:
+
+```bash
+curl -s  localhost:9898/env | grep FROM
+  "FROM_ESC_VIA_ESO=world",
+  "FROM_ESC_VIA_ESO_2=world",
+```
+
+And you should see that the secret was successfully mounted into the pod and the application is running as expected.
```

Comment:
...->"And here we see secret values from the ESC project, proxied by ESO, and passed to the `PodInfo` application as environment variables. That means the secret was successfully mounted into the pod and the application is running as expected."